consultation_analyser/consultations/models.py

```diff
@@ -40,301 +41,237 @@ class Meta:
         ordering = ["created_at"]
 
 
-class Consultation(UUIDPrimaryKeyModel, TimeStampedModel):
-    name = models.CharField(max_length=256)
-    slug = models.CharField(null=False, max_length=256)
-    users = models.ManyToManyField(User)
-
-    def has_processing_run(self):
-        return ProcessingRun.objects.filter(consultation=self).exists()
+class SlugFromTextModel(models.Model):
+    text = models.CharField(max_length=256)
+    slug = models.SlugField(null=False, editable=False, max_length=256)
 
-    class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
-        constraints = [
-            models.UniqueConstraint(fields=["slug"], name="unique_consultation_slug"),
-        ]
+    def save(self, *args, **kwargs):
+        # Generate a slug from the text - ensure unique by adding timestamp if needed.
+        # Don't allow empty slug.
+        ModelClass = self.__class__
+        cropped_length = 220
+        cropped_text = self.text[:cropped_length]
+        generated_slug = slugify(cropped_text)
+        if self.pk:
+            slug_exists = (
+                ModelClass.objects.filter(slug=generated_slug).exclude(pk=self.pk).exists()
+            )
+        else:
+            slug_exists = ModelClass.objects.filter(slug=generated_slug).exists()
```

Comment:
What does this if statement get us that `UniqueConstraint` wouldn't?

Reply:
What the `save()` does is:
* Generate slug from text
* If slug is unique - save that and we're done
* If slug is not unique - make it unique by adding a timestamp at the end (though will happen infrequently)
`UniqueConstraint` we would have to wait til we've saved to find out the slug was invalid.